plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -161,6 +164,68 @@ class="screen-reader-text"
 		return $block_content;
 	}
 
+	/**
+	 * Track feature usage of the Product Collection block within the site editor.
+	 *
+	 * @param int      $post_id  The post ID.
+	 * @param \WP_Post $post     The post object.
+	 *
+	 * @return void
+	 */
+	public function track_collection_instances( $post_id, $post ) {
+
+		if ( ! defined( 'REST_REQUEST' ) || ! REST_REQUEST || ! wc_current_theme_is_fse_theme() ) {
+			return;
+		}
+
+		if ( ! $post instanceof \WP_Post ) {
+			return;
+		}
```

Comment:
I added this check because I've seen 3PDs running a `do_action` call directly. But, similar to [this](https://github.com/woocommerce/woocommerce/pull/46466/files/d9fbf3bda2b689d8d0a2e78f1171ac69f86b4cb9#r1634887401), I think it's OK to trust the `save_post` spec.

Reply:
Got it! I agree that we can trust the `save_post` spec. But it's a nitpick, so feel free to ignore it. It's always good to have extra check if you believe there can be a case when $post won't be an instance of `\WP_Post`.

Reply:
I like this defensive programming.